apps/mobile/src/utils/analytics.ts

```diff
@@ -23,11 +26,26 @@ const leatherAnalyticsClient = configureAnalyticsClient<SegmentClient>({
 export let analytics: ReturnType<typeof configureAnalyticsClient<SegmentClient>> | undefined =
   leatherAnalyticsClient;
 
+export async function trackFirstAppOpen() {
+  try {
+    const hasTrackedFirstOpen = await AsyncStorage.getItem(FIRST_OPEN_KEY);
+
+    if (!hasTrackedFirstOpen) {
+      await analytics?.untypedTrack('application_first_opened', {
+        timestamp: new Date().toISOString(),
+      });
+      await AsyncStorage.setItem(FIRST_OPEN_KEY, 'true');
+    }
+  } catch (error) {
+    console.error('Failed to track first app open:', error);
+  }
+}
+
 store.subscribe(async () => {
   const analyticsPreference = selectAnalyticsPreference(store.getState());
 
   if (analyticsPreference === 'rejects-tracking') {
-    await analytics?.track('user_setting_updated', {
+    await analytics?.untypedTrack('user_setting_updated', {
```

Comment:
Why untyped?

Reply:
I didn't write it, I generated it! Good catch 👍

Reply:
I updated this and added the first tracked event properly